packages/domain-picker/src/components/index.tsx

```diff
@@ -400,7 +404,7 @@ const DomainPicker: FunctionComponent< Props > = ( {
 															}
 															hstsRequired={ suggestion.hsts_required }
 															isFree={ suggestion.is_free }
-															isRecommended={ isRecommended }
+															isRecommended={ hideRecommendationLabel ? false : isRecommended }
```

Comment:
In the context of https://github.com/Automattic/wp-calypso/pull/52081/files#r622168030
```suggestion
															isRecommended={ showRecommendationLabel && isRecommended }
```